src/widgets.ts

```diff
@@ -83,6 +85,39 @@ export abstract class Widget<
     return dispatchWidgetPostMessageEvent
   }
 
+  /**
+   * Public method to communicate with the iframe widget about navigation events
+   * Can be called when the host has a 'back' event happen.
+   * This will send a post message event to the iframe widget, which if it's
+   * listening for the 'mx/navigation' event, it can make its changes and then
+   * send its own post message event back with a `did_go_back` properity.
+   */
+  navigateBack() {
+    return new Promise((resolve) => {
+      const iframeElement = this.iframe.contentWindow
+
+      if (!iframeElement) {
+        throw new Error("Unable to navigate back, iframe element is not available.")
+      }
+
+      // If we get a navigation event back, resolve the promise with the value `did_go_back`
+      const handleIncomingNavigationEvent = (e: MessageEvent) => {
+        if (e.data.type === "mx/navigation") {
+          window.removeEventListener("message", handleIncomingNavigationEvent)
+          resolve(e.data.metadata.did_go_back)
+        }
+      }
+
+      // Set up temporary listener to listen for navigation events from the iframe widget
+      window.addEventListener("message", handleIncomingNavigationEvent, false)
+
+      const data = { mx: true, type: "mx/navigation", payload: { action: "back" } }
```

Comment:
Not a blocker, just a nit, but can we group the variable declarations in this method?